src/prf_expand.rs

```diff
@@ -96,6 +96,12 @@ impl PrfExpand<([u8; 32], [u8; 4])> {
     pub const SPROUT_ZIP32_CHILD: Self = Self::new(0x80);
     pub const ORCHARD_ZIP32_CHILD: Self = Self::new(0x81);
     pub const ARBITRARY_ZIP32_CHILD: Self = Self::new(0xAB);
+
+    /// Expands the given secret key in this domain, with an additional
+    /// `tag` input.
+    pub fn with_tag(self, c_par: &[u8], sk_par: &[u8; 32], i: &[u8; 4], tag: &[u8]) -> [u8; 64] {
```

Comment:
Note that `tag` is variable-length, which the `with_inputs!` macro doesn't support. This seems to be the simplest workaround, short of using `apply` directly in the `zip32` crate which would lose some type safety. We end up defining `with_tag` also for Sprout which is unintentional but harmless, since it will not be exposed in higher-level APIs.